superset-frontend/src/cccs-viz/plugins/plugin-chart-ag-grid/src/ag-grid/ContextMenu/MenuItems/RetainEmlMenuItem.tsx

```diff
@@ -0,0 +1,68 @@
+import { SupersetClient } from '@superset-ui/core';
+import React from 'react';
+import { Menu } from 'src/components/Menu';
+import { useDispatch } from 'react-redux';
+import {
+  addInfoToast,
+  addDangerToast,
+} from 'src/components/MessageToasts/actions';
+import AlfredIcon from 'src/cccs-viz/plugins/components/AlfredIcon';
+import { Tooltip } from 'antd';
+
+interface RetainEmlMenuItemProps {
+  label: string;
+  data: string[];
+  onSelection: () => void;
+  key?: string;
+  disabled?: boolean;
+  isContextMenu?: boolean;
+  contextMenuY?: number;
+  tooltip?: string;
+}
+
+export default function RetainEmlMenuItem(props: RetainEmlMenuItemProps) {
+  const dispatch = useDispatch();
+
+  const onClick = () => {
+    const endpoint = `/api/v1/fission/retain-eml-record?cbs_email_ids=${props.data}`;
```

Comment:
I wouldn't advocate for changing this pre-release, but thoughts on pulling some of these URL-type pieces into the deployment config?

Reply:
Yea we can definitely do that in the future. I am also for not changing it before the release though